request.js

```diff
@@ -577,7 +577,8 @@ Request.prototype.init = function (options) {
         }
         end()
       })
-    } else {
+    } else if (self.endRequest !== false) {
```

Comment:
what about all the other places where end() & self.end() are called? Should those be blocked from automatically closing as well?